### ISSUE TITLE
Paths storing fails

### ISSUE THREAD

**Issue by Templayer** [melonDS-emu/melonDS#2116]
![image](https://github.com/user-attachments/assets/1c99524a-7c3c-451d-85c5-33f98f49f729)

I can enter paths just fine. It prompts a restart if confirmed.
(only the ends of paths are shown, as the entire path doesn't fit the input screen) 

After restarting the application, it works as it should. Opening a savefile load dialog correctly shows the set folder.

Upon closing and reopening the application manually, however, the paths are gone.

![image](https://github.com/user-attachments/assets/520e7bee-c9ab-4b9e-8212-346f33b92d70)

Using today's nightly.

**Comment by Templayer** [comment]
[melonDS result config file.zip](https://github.com/user-attachments/files/16429122/melonDS.result.config.file.zip)

**Comment by RSDuck** [comment]
Is it possible that you have melonDS in a folder for which it doesn't have any write permissions (e.g. C:/Program files)?

**Comment by Templayer** [comment]
> Is it possible that you have melonDS in a folder for which it doesn't have any write permissions (e.g. C:/Program files)?

No. Other settings (like last ROM folder etc. - that even has diacritics) are stored correctly.

Also, no UAC, Win7. ;)

**Comment by Templayer** [comment]
(I have added the resulting config file)

**Comment by nadiaholmquist** [comment]
That config file is an INI, while latest melonDS should save to a TOML file. Are you sure you're using the newest version or did you maybe attach the wrong file?

**Comment by Templayer** [comment]
> That config file is an INI, while latest melonDS should save to a TOML file. Are you sure you're using the newest version or did you maybe attach the wrong file?

I can't see a TOML file:

![image](https://github.com/user-attachments/assets/fcda3988-8329-4195-9ef5-c0c411ef4952)

But the ini is indeed ancient - looks like I have been wrong previously.

Maybe writing fails when it is in a folder with special unicode symbols like diacritics? 

https://github.com/melonDS-emu/melonDS/issues/2113 

It would be nice if there were logs I could take a look at.

**Comment by nadiaholmquist** [comment]
Then it's not the newest build. Try https://github.com/melonDS-emu/melonDS/actions/runs/10154377746/artifacts/1753435768

**Comment by Templayer** [comment]
> Then it's not the newest build. Try https://github.com/melonDS-emu/melonDS/actions/runs/10154377746/artifacts/1753435768

That seems to be false. It has the same modified date and time as what I have used to make the previous printscreens.

![image](https://github.com/user-attachments/assets/da62debf-f34a-4402-a1c9-7b57f7e81477)

**Comment by Templayer** [comment]
> I have confirmed it is a unicode issue. Renaming the folder to remove diacritics created a toml file... but...
> 
> ![image](https://private-user-images.githubusercontent.com/11728067/353498848-a6cf7526-92cf-4e07-b9a3-4f626cb1995d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzNTAxMDEsIm5iZiI6MTcyMjM0OTgwMSwicGF0aCI6Ii8xMTcyODA2Ny8zNTM0OTg4NDgtYTZjZjc1MjYtOTJjZi00ZTA3LWI5YTMtNGY2MjZjYjE5OTVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMwVDE0MzAwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliNGM4NGM3ZTFhZjgxNjBlOTBlYzE0YjM2N2RkOTU1NmEwOTk5OGUwMTM4ZmM0YWQwMjQ2ZTE5NThmMGY1ZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OzHvayg1Swl4el_4-BLHUdn_F70QTcfu1l-oZu92wPQ)

Renaming the folder back doesn't make it not crash - it crashes when it attempts to read the toml file. Deleting it in a folder when creating it fails (putting diacritics back) makes it run again.

So saving a toml file has two issues:
A) it fails when it attempts to write into a folder with special unicode symbols such as diacritics (you can't really have Czech without diacritics...)
B) it fails to read the toml file for reasons unknown. The file doesn't appear to be corrupted, but I have already deleted it as I need the emulator to be working for tomorrow.

**Comment by Templayer** [comment]
[melonDS.toml.zip](https://github.com/user-attachments/files/16434123/melonDS.toml.zip)

This toml file refuses to load for some reason, crashing the emulator.

So far I found a **workaround** - I manually edited my original configuration file, which is still read correctly.

(unless the issue is the "conversion" of the old ini file by loading its values into memory and writing that toml file - the ini file has been uploaded previously)

`3DRenderer=2` in the ini file set the renderer to OpenGL (Compute shader), so that's nice. I don't have to set that up every time.

I did fail to to set the save and savestate folders, though. These variations are not loaded by the emulator from the ini file:

```
SaveFilePath=V:/Melon DS emulátor/SAVY
SavestatePath=V:/Melon DS emulátor/SAVY

SaveFilePath=V:/Melon DS emulátor/SAVY/
SavestatePath=V:/Melon DS emulátor/SAVY/

SaveFilePath=V:\Melon DS emulátor\SAVY\
SavestatePath=V:\Melon DS emulátor\SAVY\
```

THESE, however, were already there, and they are loaded correctly:

```
BIOS9Path=V:/Melon DS emulátor/BIOS and firmware/biosnds9.bin
BIOS7Path=V:/Melon DS emulátor/BIOS and firmware/biosnds7.bin
FirmwarePath=V:/Melon DS emulátor/BIOS and firmware/dsfirmware.bin
LastROMFolder=J:/VIDEOSOUBORY NAHRAVANO/Skineri Hraje Pokémony/DALŠÍ HRY
RecentROM_0=J:/VIDEOSOUBORY NAHRAVANO/Skineri Hraje Pokémony/DALŠÍ HRY/Pokemon Sacred Gold v1.1.nds
```

Video evidence: https://www.youtube.com/watch?v=eR0-wmltrEo

**Comment by nadiaholmquist** [comment]
Download the latest build, Generic should have fixed the TOML file saving issue.
